synapse/handlers/saml_handler.py

```diff
@@ -120,19 +139,39 @@ def handle_redirect_request(self, client_redirect_url):
             logger.warning("SAML2 response lacks a 'uid' attestation")
             raise SynapseError(400, "uid not in SAML2 response")
 
-        try:
-            mxid_source = saml2_auth.ava[self._mxid_source_attribute][0]
-        except KeyError:
-            logger.warning(
-                "SAML2 response lacks a '%s' attestation", self._mxid_source_attribute
-            )
-            raise SynapseError(
-                400, "%s not in SAML2 response" % (self._mxid_source_attribute,)
-            )
-
         self._outstanding_requests_dict.pop(saml2_auth.in_response_to, None)
 
-        displayName = saml2_auth.ava.get("displayName", [None])[0]
+        # Map saml response to user attributes using the configured mapping provider
+        for i in range(1000):
+            try:
+                attribute_dict = self._user_mapping_provider.saml_response_to_user_attributes(
+                    self._saml2_user_mapping_provider_config, saml2_auth, i
+                )
+            except Exception:
+                logging.exception("Error in SAML mapping provider plugin")
+                raise SynapseError(500, "Error parsing SAML2 response")
```

Comment:
I'd be tempted just to let the exception propagate, but :man_shrugging:

Reply:
I suppose it does effectively do that.